lib/spack/spack/util/path.py

```diff
@@ -204,12 +204,8 @@ def format_os_path(path, mode=Path.unix):
         mode (Path): the path filesperator style to normalize the
             passed path to. Default is unix style, i.e. '/'
     """
-    if not path:
-        return path
```

Comment:
My guess is that these two lines were playing a role in the test that is now failing. I wonder why, since if we pass in only strings, like the doc seems to say, then the only falsish string is the empty string, and that is preserved. I would be different if we pass in other kind of input e.g. `None`.